src/streams/send.rs

```diff
@@ -696,6 +703,28 @@ impl StreamTx {
 
         let mut snapshot = self.send_queue.snapshot(now);
 
+        // Outstanding resends
+        let mut resend_snapshot =
+            self.resends
+                .iter()
+                .fold(QueueSnapshot::default(), |mut snapshot, r| {
+                    snapshot.total_queue_time_origin += now.duration_since(r.queued_at);
+                    snapshot.size += r.payload_size;
+                    snapshot.packet_count += 1;
+                    snapshot.first_unsent = snapshot
+                        .first_unsent
+                        .map(|i| i.min(r.queued_at))
+                        .or(Some(r.queued_at));
+
+                    snapshot
+                });
+        resend_snapshot.created_at = now;
+        if resend_snapshot.packet_count > 0 {
+            resend_snapshot.priority = QueuePriority::Media;
+        }
```

Comment:
Can you move this calculation to its own function? Just for tidyness.

Reply:
Yep I was weighing that already